docs/rules/unused-prop-types.md

```diff
@@ -0,0 +1,70 @@
+# Prevent definitions of unused prop types (unused-prop-types)
+
+Warns you if you have defined a prop type but it is never being used anywhere.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+var Hello = React.createClass({
+  propTypes: {
+    name: React.PropTypes.string
+  },
+  render: function() {
+    return <div>Hello Bob</div>;
+  }
+});
+
+var Hello = React.createClass({
+  propTypes: {
+    firstname: React.PropTypes.string.isRequired,
+    middlename: React.PropTypes.string.isRequired, // middlename is never used below
+	lastname: React.PropTypes.string.isRequired
+  },
+  render: function() {
+    return <div>Hello {this.props.firstname} {this.props.lastname}</div>;
+  }
+});
+
+function Hello({ name }) {
```

Comment:
I don't understand why this is included in the examples--this component has no propTypes. Did you mean to include something like this?

```js
Hello.propTypes = {
  name: React.PropTypes.string.isRequired,
};
```